libpromises/evalfunction.c

```diff
@@ -4860,25 +4861,50 @@ static FnCallResult FnCallFormat(EvalContext *ctx, ARG_UNUSED const Policy *poli
 
 static FnCallResult FnCallIPRange(EvalContext *ctx, ARG_UNUSED const Policy *policy, ARG_UNUSED const FnCall *fp, const Rlist *finalargs)
 {
+    const bool iprange_mode = (strcmp(fp->name, "iprange") == 0);
     char *range = RlistScalarValue(finalargs);
 
     if (!FuzzyMatchParse(range))
     {
         return FnFailure();
     }
 
-    for (const Item *ip = EvalContextGetIpAddresses(ctx); ip != NULL; ip = ip->next)
+    /* Case 1: iprange(): No second argument, just arg1: check arg1 (a range) against VIPADDRESS */
+    if (iprange_mode && finalargs->next == NULL)
     {
-        Rlist *ifaces = finalargs->next;
-        // we match on VIPADDRESS iff no interfaces were requested
-        if (FuzzySetMatch(range, VIPADDRESS) == 0 && NULL == ifaces)
+        if (FuzzySetMatch(range, VIPADDRESS) == 0)
         {
-            Log(LOG_LEVEL_DEBUG, "%s: found range %s on hostip %s, no interface", fp->name, range, ip->name);
+            Log(LOG_LEVEL_DEBUG, "%s: found range %s on hostip %s, no interface", fp->name, range, VIPADDRESS);
             return FnReturnContextTrue();
         }
-        else if (FuzzySetMatch(range, ip->name) == 0)
+
+        return FnReturnContextFalse();
```

Comment:
This alters old behaviour: it used to search for a match on all interfaces in case VIPADDRESS didn't match.

Ugh, this function is way too complicated, and I thought it should be easy to because of the few LOC...